src/virtual-machine.js

```diff
@@ -440,6 +440,31 @@ class VirtualMachine extends EventEmitter {
     }
 
     /**
+     * Duplicate a sprite.
+     * @param {string} targetId ID of a target whose sprite to duplicate.
+     */
+    duplicateSprite (targetId) {
+        const target = this.runtime.getTargetById(targetId);
+
+        if (target) {
+            if (!target.isSprite()) {
+                throw new Error('Cannot duplicate non-sprite targets.');
+            }
+            const sprite = target.sprite;
+            if (!sprite) {
+                throw new Error('No sprite associated with this target.');
+            }
+
+            target.duplicate().then(newTarget => {
+                this.runtime.targets.push(newTarget);
+                this.setEditingTarget(newTarget.id);
+            });
+        } else {
+            throw new Error('No target with the provided id.');
```

Comment:
`if (target)` at the top makes sense if you're trying to avoid early-outs, but then there are two early-out `throw`s inside... I suggest sticking to one pattern or the other. Personally, I'd move this `throw` into an `if (!target)` near the top and un-indent the current body of `if (target)` but I'm OK with either pattern.

Reply:
Good call, I was kind of blindly working from the other methods in this file. Refactored in a885e7d